var/spack/repos/builtin/packages/llvm/package.py

```diff
@@ -148,32 +154,32 @@ class Llvm(CMakePackage):
     # https://bugs.llvm.org/show_bug.cgi?id=39696
     patch('thread-p9.patch', when='@develop+libcxx')
 
-    @run_before('cmake')
-    def check_darwin_lldb_codesign_requirement(self):
-        if not self.spec.satisfies('+lldb platform=darwin'):
```

Comment:
Just for curiosity. Why was this part of the code ok using `self` and not the next one?

Reply:
Because it's inside of a instance method that provides self as a parameter.